### ISSUE TITLE
compiler segmentation fault

### ISSUE THREAD

**Issue by xBZZZZ** [ziglang/zig#18392]
### Zig Version

0.12.0-dev.1856+94c63f31f

### Steps to Reproduce and Observed Behavior

```bash
echo 'export fn asd() void{
_=@extern(*anyopaque,.{.name="a"});
_=@extern(*const fn() callconv(.C) usize,.{.name="b"})();
}' > bug.zig

zig build-obj bug.zig #Segmentation fault (core dumped)
```

### Expected Behavior

`bug.o` and `bug.o.o` files created (I don't know why `zig build-obj` creates 2 *.o files when success)

## edit: simpler code that segmentation fault compiler
```zig
export fn asd() void{
	_=@extern(*anyopaque,.{.name="a"});
	@extern(*const fn() callconv(.C) void,.{.name="b"})();
}
```

**Comment by nektro** [comment]
`0.12.0-dev.1836+dd189a354`

```
thread 14024 panic: InternPool.getCoerced of variable not implemented from ptr_type to ptr_type
/home/meghan/src/zig/lib/std/debug.zig:373:22: 0x1f117bc in panicExtra__anon_129826 (zig)
    std.builtin.panic(msg, trace, ret_addr);
                     ^
/home/meghan/src/zig/lib/std/debug.zig:348:15: 0x1a28766 in panic__anon_111011 (zig)
    panicExtra(null, null, format, args);
              ^
/home/meghan/src/zig/src/InternPool.zig:7043:24: 0x147c114 in getCoerced (zig)
        std.debug.panic("InternPool.getCoerced of {s} not implemented from {s} to {s}", .{
                       ^
/home/meghan/src/zig/src/Module.zig:5656:62: 0x105294e in getCoerced (zig)
    return Value.fromInterned((try mod.intern_pool.getCoerced(mod.gpa, val.toIntern(), new_ty.toIntern())));
                                                             ^
/home/meghan/src/zig/src/value.zig:270:62: 0xdb6d48 in intern (zig)
        if (val.ip_index != .none) return (try mod.getCoerced(val, ty)).toIntern();
                                                             ^
/home/meghan/src/zig/src/Module.zig:642:45: 0xdb61ff in internValue (zig)
        const ip_index = try decl.val.intern(decl.ty, mod);
                                            ^
/home/meghan/src/zig/src/Module.zig:5579:29: 0xda6a39 in markDeclAlive (zig)
    _ = try decl.internValue(mod);
                            ^
/home/meghan/src/zig/src/codegen/llvm.zig:4531:30: 0x10a0059 in lowerDeclRefValue (zig)
        try mod.markDeclAlive(decl);
                             ^
/home/meghan/src/zig/src/codegen/llvm.zig:3878:60: 0xde2bba in lowerValue (zig)
                    .decl => |decl| try o.lowerDeclRefValue(ptr_ty, decl),
                                                           ^
/home/meghan/src/zig/src/codegen/llvm.zig:4809:42: 0x187f4a1 in resolveValue (zig)
        const llvm_val = try o.lowerValue(tv.val.toIntern());
                                         ^
/home/meghan/src/zig/src/codegen/llvm.zig:4798:47: 0x187f1b7 in resolveInst (zig)
        const llvm_val = try self.resolveValue(.{
                                              ^
/home/meghan/src/zig/src/codegen/llvm.zig:5120:45: 0x18c65a2 in airCall (zig)
        const llvm_fn = try self.resolveInst(pl_op.operand);
                                            ^
/home/meghan/src/zig/src/codegen/llvm.zig:4985:55: 0x135ab73 in genBody (zig)
                .call              => try self.airCall(inst, .auto),
                                                      ^
/home/meghan/src/zig/src/codegen/llvm.zig:1648:19: 0x13540f5 in updateFunc (zig)
        fg.genBody(air.getMainBody()) catch |err| switch (err) {
                  ^
/home/meghan/src/zig/src/link/Elf.zig:3174:70: 0x1367cd6 in updateFunc (zig)
    if (self.llvm_object) |llvm_object| return llvm_object.updateFunc(mod, func_index, air, liveness);
                                                                     ^
/home/meghan/src/zig/src/link.zig:624:77: 0x101abc2 in updateFunc (zig)
            .elf   => return @fieldParentPtr(Elf,   "base", base).updateFunc(module, func_index, air, liveness),
                                                                            ^
/home/meghan/src/zig/src/Module.zig:3390:37: 0xd99c7a in ensureFuncBodyAnalyzed (zig)
            comp.bin_file.updateFunc(mod, func_index, air, liveness) catch |err| switch (err) {
                                    ^
/home/meghan/src/zig/src/Compilation.zig:3761:42: 0xd9762d in processOneJob (zig)
            module.ensureFuncBodyAnalyzed(func) catch |err| switch (err) {
                                         ^
/home/meghan/src/zig/src/Compilation.zig:3698:30: 0xb69528 in performAllTheWork (zig)
            try processOneJob(comp, work_item, main_progress_node);
                             ^
/home/meghan/src/zig/src/Compilation.zig:2505:31: 0xb64d82 in update (zig)
    try comp.performAllTheWork(main_progress_node);
                              ^
/home/meghan/src/zig/src/main.zig:4283:24: 0xb95482 in updateModule (zig)
        try comp.update(main_progress_node);
                       ^
/home/meghan/src/zig/src/main.zig:3684:17: 0xbb8042 in buildOutputType (zig)
    updateModule(comp) catch |err| switch (err) {
                ^
/home/meghan/src/zig/src/main.zig:282:31: 0x9c17ef in mainArgs (zig)
        return buildOutputType(gpa, arena, args, .{ .build = .Obj });
                              ^
/home/meghan/src/zig/src/main.zig:222:20: 0x9be8c5 in main (zig)
    return mainArgs(gpa, arena, args);
                   ^
/home/meghan/src/zig/lib/std/start.zig:585:37: 0x9be346 in main (zig)
            const result = root.main() catch |err| {
                                    ^
???:?:?: 0x7f8bd783dacd in ??? (libc.so.6)
Unwind information for `libc.so.6:0x7f8bd783dacd` was not available, trace may be incomplete

Aborted (core dumped)
```

the other issue mentioned is https://github.com/ziglang/zig/issues/13179

**Comment by Vexu** [comment]
Fixed by #18554
